src/blazar_tempest_plugin/tests/scenario/test_images.py

```diff
@@ -152,20 +152,25 @@ def resource_setup(cls):
                     reservation_id = inst._get_host_reservation(lease)
                     flavor = CONF.reservation.reservable_flavor_ref
                     scheduler_hints = {"reservation": reservation_id}
-                else:
-                    flavor = CONF.compute.flavor_ref
+                elif CONF.reservation.reservation_type == "kvm":
+                    flavor_id = inst._get_flavor_id(CONF.reservation.reservable_flavor_ref)
+                    lease = inst._reserve_flavor_host(flavor_id=flavor_id)
+                    cls.lease_id = lease["id"]
+                    reservation_id = inst._get_flavor_reservation(lease)
                     scheduler_hints = {}
+                    flavor = f"reservation:{reservation_id}"
+                else:
+                    raise Exception(f"Invalid reservation_type: {CONF.reservation.reservation_type}")
```

Comment:
Although our current sites all require reservations, I think it's still useful for this to work in the case where reservations aren't required.

Reply:
Do you have more details on what that would look like? Should it work on bare metal without reservations, KVM, both?